include/shardwrapper.hpp

```diff
@@ -153,10 +154,17 @@ template <class T> struct ShardWrapper {
 
     // setParam
     if constexpr (has_setParam<T>::value) {
-      result->setParam = static_cast<SHSetParamProc>(
-          [](Shard *b, int i, const SHVar *v) { reinterpret_cast<ShardWrapper<T> *>(b)->shard.setParam(i, *v); });
+      result->setParam = static_cast<SHSetParamProc>([](Shard *b, int i, const SHVar *v) {
```

Comment:
Does this already cover set/get params we talked about? https://github.com/fragcolor-xyz/shards/issues/265

Reply:
Yeah I implemented error handling for set/get yep